Sources/BraintreeCoreSwift/BTAnalyticsService.swift

```diff
@@ -0,0 +1,174 @@
+        }
+    }
+
+    func flush(_ completion: @escaping (Error?) -> Void = { _ in }) {
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            guard let configuration, error == nil else {
+                if let error {
+                    completion(error)
+                }
+                return
+            }
+
+            guard let analyticsURL = configuration.json?["analytics"]["url"].asURL() else {
+                completion(BTAnalyticsServiceError.missingAnalyticsURL)
+                return
+            }
+
+            if self.http == nil {
+                if let clientToken = self.apiClient.clientToken {
+                    self.http = BTHTTP(url: analyticsURL, authorizationFingerprint: clientToken.authorizationFingerprint)
+                } else if let tokenizationKey = self.apiClient.tokenizationKey {
+                    self.http = BTHTTP(url: analyticsURL, tokenizationKey: tokenizationKey)
+                } else {
+                    completion(BTAnalyticsServiceError.invalidAPIClient)
+                    return
+                }
+            }
+
+            // A special value passed in by unit tests to prevent BTHTTP from actually posting
+            if self.http?.baseURL.absoluteString == "test://do-not-send.url" {
+                completion(nil)
+                return
+            }
+
+            self.sessionsQueue.async {
+                if self.analyticsSessions.count == 0 {
+                    completion(nil)
+                    return
+                }
+
+                var willPostAnalyticsEvent: Bool = false
+
+                self.analyticsSessions.keys.forEach { sessionID in
+                    var session = self.analyticsSessions[sessionID]
+
+                    willPostAnalyticsEvent = true
+
+                    let metadataParameters: [String: Any] = [
+                        "sessionId": session?.sessionID ?? "",
+                        "integrationType": session?.integration ?? "",
+                        "source": session?.source ?? ""
+                    ]
+                        .merging(session?.metadataParameters ?? [:]) { $1 }
+
+                    var postParameters: [String: Any] = [:]
+
+                    if let sessionEvents = session?.events {
+                        // Map array of BTAnalyticsEvent to JSON
+                        postParameters["analytics"] = sessionEvents.map { $0.json }
+                    }
+
+                    postParameters["_meta"] = metadataParameters
+
+                    if let authorizationFingerprint = self.apiClient.clientToken?.authorizationFingerprint {
+                        postParameters["authorization_fingerprint"] = authorizationFingerprint
+                    } else if let tokenizationKey = self.apiClient.tokenizationKey {
+                        postParameters["tokenization_key"] = tokenizationKey
+                    }
+
+                    session?.events.removeAll()
```

Comment:
Could it make sense to factor this out into a method called something like `createAnalyticsEvent`?